data_prep/project_src.py

```diff
@@ -271,7 +271,15 @@ def _copy_project_src_from_local(project: str, out: str):
     logging.error('Failed to copy /src from OSS-Fuzz image of %s:', project)
     logging.error('STDOUT: %s', result.stdout)
     logging.error('STDERR: %s', result.stderr)
+
+    # Shut down the container that was just started before throwing exception.
+    sp.run(['docker', 'container', 'stop', f'{project}-container'])
```

Comment:
Can we do a `try ... finally` here and put the stop in in the `finally` ?

Reply:
+1

Reply:
As there's no `try-except` in the first place I just wanted to double check -- do you suggest wrapping the initial `result = sp.run(copy_src, capture_output=True, stdin=sp.DEVNULL, check=False)` in a try?

Reply:
Maybe something like this so that we only need one `sp.run(['docker', 'container', 'stop', f'{project}-container'], check=True)`?
```python
try:
    result = sp.run(copy_src, capture_output=True, stdin=sp.DEVNULL, check=False)
    if result.returncode:
        logging.error('Failed to copy /src from OSS-Fuzz image of %s:', project)
        logging.error('STDOUT: %s', result.stdout.decode())
        logging.error('STDERR: %s', result.stderr.decode())
        raise Exception('Copy operation failed')
finally:
    sp.run(['docker', 'container', 'stop', f'{project}-container'], check=True)
```

Reply:
Done, thanks!

Reply:
Thanks for fixing this, the report also looks great.

I added [377aab2](https://github.com/google/oss-fuzz-gen/pull/311/commits/377aab2d86e71133de8ba5bd18ae61f5c5d33b6c) to:
1. Avoid weird failure when `stdin=sp.DEVNULL` is not set; we noticed some of this in the past.
2. Log errors if the stop command fails.
3. Set `check=False` and `capture_output=True` to make logging consistent with other `sp` calls above.

Reply:
> Thanks for fixing this, the report also looks great.
> 
> I added [377aab2](https://github.com/google/oss-fuzz-gen/pull/311/commits/377aab2d86e71133de8ba5bd18ae61f5c5d33b6c) to:
> 
> 1. Avoid weird failure when `stdin=sp.DEVNULL` is not set; we noticed some of this in the past.
> 2. Log errors if the stop command fails.
> 3. Set `check=False` and `capture_output=True` to make logging consistent with other `sp` calls above.
> 
> Hope you don't mind : )

For bug fixes I would faster landing and follow up PRs. This makes it easier to use the fix for continued dev.